src/generate_pisa_index.cpp

```diff
@@ -21,19 +41,138 @@ std::ostream &write_sequence(std::ostream &os, gsl::span<T> sequence)
 }
 
 
+struct inverted_index {
+
+  std::unordered_map<uint32_t, std::vector<uint32_t>> m_documents {};
+  std::unordered_map<uint32_t, std::vector<uint32_t>> m_frequencies {};
```

Comment:
Correct me if I'm wrong, but I think you're just using consecutive numbers for term IDs. Why not `std::vector` and just push each new posting list to the end? Then you wouldn't even need to pass the term to `add_document_lengths`, but instead simply start with `auto term_id = static_cast<uint32>(m_documents.size())` before pushing the new one, or something like this.

Reply:
Okay, agreed. I'll refactor.